pytest/tests/mocknet/sharded_bm.py

```diff
@@ -0,0 +1,311 @@
+    base_genesis_patch = f"{REMOTE_BENCHNET_DIR}/{args.case}/{args.bm_params['base_genesis_patch']}"
+
+    config = f"{NEAR_HOME}/config.json"
+    base_config_patch = f"{REMOTE_BENCHNET_DIR}/{args.case}/{args.bm_params['base_config_patch']}"
+    config_patch = f"{REMOTE_BENCHNET_DIR}/{args.case}/config_patch.json"
+
+    log_config = f"{NEAR_HOME}/log_config.json"
+    log_config_patch = f"{REMOTE_BENCHNET_DIR}/cases/log_patch.json"
+
+    run_cmd_args = copy.deepcopy(args)
+    run_cmd_args.cmd = f"\
+        python3 {REMOTE_BENCHNET_DIR}/helpers/json_updater.py {genesis} {base_genesis_patch} \
+        && python3 {REMOTE_BENCHNET_DIR}/helpers/json_updater.py {config} {base_config_patch} {config_patch} \
+        && touch {log_config} \
+        && python3 {REMOTE_BENCHNET_DIR}/helpers/json_updater.py {log_config} {log_config_patch} \
+    "
+
+    if tracing_server_ip is None:
+        run_cmd_args.cmd += f"\
+            && jq '.opentelemetry = null' {log_config} >tmp.$$.json && mv tmp.$$.json {log_config} || rm tmp.$$.json \
+        "
+
+    run_remote_cmd(CommandContext(run_cmd_args))
+
+    start_nodes(args)
+
+    time.sleep(10)
+
+    run_cmd_args = copy.deepcopy(args)
+    run_cmd_args.host_filter = f"({'|'.join(args.forknet_details['cp_instance_names'])})"
+    accounts_path = f"{REMOTE_BENCHNET_DIR}/user-data/shard.json"
+    run_cmd_args.cmd = f"\
+        shard=$(python3 {REMOTE_BENCHNET_DIR}/helpers/get_tracked_shard.py) && \
+        echo \"Tracked shard: $shard\" && \
+        rm -rf {REMOTE_BENCHNET_DIR}/user-data && \
+        mkdir -p {REMOTE_BENCHNET_DIR}/user-data && \
+        cp {NEAR_HOME}/user-data/shard_$shard.json {accounts_path} \
+    "
+
+    run_remote_cmd(CommandContext(run_cmd_args))
+
+    handle_stop(args)
+
+
+def handle_stop(args):
+    """Handle the stop command - stop the benchmark."""
+    logger.info("Stopping nodes")
+    stop_nodes_cmd_args = copy.deepcopy(args)
+    stop_nodes_cmd_args.host_filter = f"({'|'.join(args.forknet_details['cp_instance_names'])})"
+    stop_nodes_cmd(CommandContext(stop_nodes_cmd_args))
+
+
+def handle_reset(args):
+    """Handle the reset command - reset the benchmark state."""
+    logger.info("Resetting benchmark state")
+    handle_stop(args)
+
+    run_cmd_args = copy.deepcopy(args)
+    config = f"{NEAR_HOME}/config.json"
+    run_cmd_args.cmd = f"\
+        find {NEAR_HOME}/data -mindepth 1 -delete && \
+        rm -rf {REMOTE_BENCHNET_DIR} && \
+        jq 'del(.tx_generator)' {config} > tmp.$$.json && mv tmp.$$.json {config} || rm tmp.$$.json \
+    "
+
+    run_remote_cmd(CommandContext(run_cmd_args))
+
+    env_cmd_args = copy.deepcopy(args)
+    env_cmd_args.clear_all = True
+    run_env_cmd(CommandContext(env_cmd_args))
+
+    reset_cmd_args = copy.deepcopy(args)
+    reset_cmd_args.backup_id = "start"
+    reset_cmd_args.yes = True
+    reset_cmd(CommandContext(reset_cmd_args))
+
+
+def start_nodes(args, with_tx_generator=False):
```

Comment:
What would be the new commands to:

- stop tx generator but keep nodes running without delete state or anything
- tweak config (quickly, without doing reset - init cycle)

Reply:
* Added flag `--disable-tx-generator` to `stop`
* Added command `apply-json-patches` (it was always about all json-s, not just config)